pkg/sql/resolver.go

```diff
@@ -703,6 +707,46 @@ func newInternalLookupCtx(descs []catalog.Descriptor, prefix *dbdesc.Immutable)
 			schemaDescs[desc.GetID()] = desc
 		}
 	}
+
+	if len(typDescs) > 0 {
+		// Since we just scanned all the descriptors, we already have everything
+		// we need to hydrate our types. Set up an accessor for the type hydration
+		// method to look into the scanned set of descriptors.
+		typeLookup := func(ctx context.Context, id descpb.ID) (tree.TypeName, catalog.TypeDescriptor, error) {
+			typDesc := typDescs[id]
+			dbDesc := dbDescs[typDesc.ParentID]
+			// We don't use the collection's ResolveSchemaByID method here because
+			// we already have all of the descriptors. User defined types are only
+			// members of the public schema or a user defined schema, so those are
+			// the only cases we have to consider here.
+			var scName string
+			switch typDesc.ParentSchemaID {
+			case keys.PublicSchemaID:
+				scName = tree.PublicSchema
+			default:
+				scName = schemaDescs[typDesc.ParentSchemaID].Name
+			}
+			name := tree.MakeNewQualifiedTypeName(dbDesc.GetName(), scName, typDesc.GetName())
+			return name, typDesc, nil
+		}
+		// Now hydrate all table descriptors.
+		for i := range descs {
+			desc := descs[i]
+			if tblDesc, ok := desc.(*tabledesc.Immutable); ok {
+				if err := typedesc.HydrateTypesInTableDescriptor(
+					ctx,
+					tblDesc.TableDesc(),
+					typedesc.TypeLookupFunc(typeLookup),
+				); err != nil {
+					// If we ran into an error hydrating the types, that means that we
+					// have some sort of corrupted descriptor state. Rather than disable
+					// uses of GetAllDescriptors, just log the error.
```

Comment:
this comment is out of date. I also think that in this case we should return an error because that means the backup is corrupted.

Reply:
Done.